File: src/main/java/com/history/tree/repositories/PersonRepository.java
package com.history.tree.repositories;

import com.history.tree.model.Person;
import org.springframework.data.r2dbc.repository.R2dbcRepository;
import org.springframework.lang.NonNull;
import org.springframework.stereotype.Repository;
import reactor.core.publisher.Flux;

@Repository
public interface PersonRepository extends R2dbcRepository<Person, Long> {

    @NonNull
    Flux<Person> findAllByTreeId(@NonNull long treeId);
}

File: src/main/java/com/history/tree/repositories/PersonRepository.kt
package com.history.tree.repositories

import com.history.tree.model.Person
import kotlinx.coroutines.flow.Flow
import org.springframework.data.repository.kotlin.CoroutineCrudRepository
import org.springframework.stereotype.Repository

@Repository
interface PersonRepository : CoroutineCrudRepository<Person, Long> {

    suspend fun findAllByTreeId(treeId: Long): Flow<Person>
}